src/main/java/engine/FormplayerArchiveFileRoot.java

```diff
@@ -0,0 +1,53 @@
+package engine;
+
+import org.commcare.modern.reference.ArchiveFileReference;
+import org.commcare.modern.reference.ArchiveFileRoot;
+import org.javarosa.core.reference.InvalidReferenceException;
+import org.javarosa.core.reference.Reference;
+import org.javarosa.core.util.PropertyUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.redis.core.ListOperations;
+import org.springframework.data.redis.core.StringRedisTemplate;
+
+import javax.annotation.Resource;
+import java.io.IOException;
+import java.util.zip.ZipFile;
+
+/**
+ * Created by willpride on 4/25/17.
+ */
+public class FormplayerArchiveFileRoot extends ArchiveFileRoot {
+
+    @Autowired
+    private StringRedisTemplate redisTemplate;
+
+    @Resource(name = "redisTemplate")
+    private ListOperations<String, String> listOperations;
```

Comment:
why does this use listOperations. we shouldn't need to use any special list stuff if all we're doing is setting a key and retrieving a key